### DISCUSSION TITLE
Not able to build the project

### DISCUSSION THREAD

**Discussion by GirishKumarSharma** [wailsapp/wails#938] [General]
Here is my environment info on my home learning PC :
go version go1.17 linux/amd64
npm version 7.5.2
node version v12.22.5
OS : Ubuntu 21.10
Wails : v1.16.8
wails init worked without any error, but when I says wails build, I got this error message : 

(base) girish@girish-System-Product-Name:~$ wails build
Wails v1.16.8 - Building Application

Error: unable to find 'project.json'. Please check you are in a Wails project directory
(base) girish@girish-System-Product-Name:~$ cd prj2/
(base) girish@girish-System-Product-Name:~/prj2$ wails build
Wails v1.16.8 - Building Application

⣷ Ensuring frontend dependencies are up to date (This may take a while)npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: my-app@0.0.0
npm ERR! Found: @angular/core@8.0.3
npm ERR! node_modules/@angular/core
npm ERR!   @angular/core@"~8.0.1" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer @angular/core@"8.2.14" from @angular/animations@8.2.14
npm ERR! node_modules/@angular/animations
npm ERR!   @angular/animations@"^8.0.2" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
npm ERR! 
npm ERR! See /home/girish/.npm/eresolve-report.txt for a full report.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/girish/.npm/_logs/2021-11-13T13_50_54_832Z-debug.log

✗ Ensuring frontend dependencies are up to date (This may take a while)
Error: exit status 1

Kindly let me know, if I need to provide more info to get the solution.

**Comment by GirishKumarSharma** [comment]
I don't know how it worked but today morning (after shutting down the PC) when I said : 

**npm config set legacy-peer-deps true**

and then : 
(base) girish@girish-System-Product-Name:~/prj2$ wails build
Wails v1.16.8 - Building Application

✓ Ensuring frontend dependencies are up to date (This may take a while)
✓ Building frontend...
✓ Ensuring Dependencies are up to date...
✓ Packing + Compiling project...
Awesome! Project 'prj2' built!
(base) girish@girish-System-Product-Name:~/prj2$ ./build/prj2 
Overriding existing handler for signal 10. Set JSC_SIGNAL_FOR_GC if you want WebKit to use a different signal
(base) girish@girish-System-Product-Name:~/prj2$ 

It got working. Now, I will explore more.

**Reply by leaanthony** [reply]
Do a search for angular in the issues and forums 👍

**Reply by GirishKumarSharma** [reply]
> Do a search for angular in the issues and forums +1

Yes, you are right. Will you please tell me the correct meaning of angular, npm, node.js, react words in your understanding, because I think without understanding the correct meaning and relationship, order, I will remain in dark for ever.
